src/hotspot/cpu/arm/register_arm.hpp

```diff
@@ -94,25 +74,88 @@ typedef VMRegImpl* VMReg;
 #define ALIGN_WIDE_ARGUMENTS 1
 #endif
 
-#define R0     ((Register)0)
-#define R1     ((Register)1)
-#define R2     ((Register)2)
-#define R3     ((Register)3)
-#define R4     ((Register)4)
-#define R5     ((Register)5)
-#define R6     ((Register)6)
-#define R7     ((Register)7)
-#define R8     ((Register)8)
-#define R9     ((Register)9)
-#define R10    ((Register)10)
-#define R11    ((Register)11)
-#define R12    ((Register)12)
-#define R13    ((Register)13)
-#define R14    ((Register)14)
-#define R15    ((Register)15)
-
-
-#define FP     ((Register)FP_REG_NUM)
+class Register {
+ private:
+  int _encoding;
+
+  constexpr explicit Register(int encoding) : _encoding(encoding) {}
+
+ public:
+  enum {
+    number_of_registers = 16,
+    max_slots_per_register = 1 << (LogBytesPerWord - LogBytesPerInt) // LogBytesPerWord depends on _LP64
```

Comment:
ARM32 is only 32-bit, so we can skip any _LP64-based computations, and just do the literal constant.